sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -156,6 +163,115 @@ internal class ViewAnnotationManagerImpl(
       }
     }
 
+  override fun cameraForAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?
+  ): CameraOptions? {
+    if (mapboxMap.style?.getProjection()?.name == ProjectionName.GLOBE || annotations.isEmpty()) {
+      return null
+    }
+    val viewAnnotationOptions = annotations.mapNotNull {
+      if (it.isVisible) return@mapNotNull getViewAnnotationOptionsByView(it) else null
+    }.filter { it.visible != false }
+
+    if (viewAnnotationOptions.isEmpty()) return null
+    val coordinates = coordinatesFromAnnotations(viewAnnotationOptions)
+    val paddings = calculateEdgeInsets(viewAnnotationOptions, edgeInsets)
+    return mapboxMap.cameraForCoordinates(
+      coordinates,
+      paddings,
+      bearing,
+      pitch
+    )
+  }
+
+  /**
+   * Function to get coordinates from list of [ViewAnnotationOptions].
+   */
+  private fun coordinatesFromAnnotations(annotationOptions: List<ViewAnnotationOptions>): List<Point> {
+    val coordinatesList = mutableListOf<Point>()
+    annotationOptions.forEach {
+      it.geometry?.let { geometry ->
+        coordinatesList.add(Point.fromJson(geometry.toJson()))
+      }
+    }
+    return coordinatesList
+  }
+
+  /**
+   * Calculate paddings to show viewAnnotations.
+   * Get the topMost, leftMost, rightMost, bottomMost annotation options and apply paddings accordingly.
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun calculateEdgeInsets(
+    viewAnnotationOptions: List<ViewAnnotationOptions>,
+    edgeInsets: EdgeInsets? = null
+  ): EdgeInsets {
+    val filteredViewAnnotations = viewAnnotationOptions.filter { it.geometry != null }
+    val topAnnotation =
+      filteredViewAnnotations.maxByOrNull { (it.geometry!! as Point).latitude() }
+    val bottomAnnotation =
+      filteredViewAnnotations.minByOrNull { (it.geometry!! as Point).latitude() }
+    val leftAnnotation =
+      filteredViewAnnotations.minByOrNull { (it.geometry!! as Point).longitude() }
+    val rightAnnotation =
+      filteredViewAnnotations.maxByOrNull { (it.geometry!! as Point).longitude() }
+
+    return EdgeInsets(
+      (edgeInsets?.top ?: 0).toDouble()
+        .plus(abs(getViewAnnotationOptionsFrame(topAnnotation)?.top ?: 0)),
```

Comment:
What will happen if topmost annotation is small (so getViewAnnotationOptionsFrame will also return some small value), but another one, that is below - is much bigger. Will edge insets calculated correctly then? Won't this bigger annotation be hidden?

Reply:
> If another one (let's say annotation2 whose geometry is below annotation1) is bigger and its top is bigger than annotation1, then annotation2 will become topmost from the calculation

@ank27 not sure I see how this is implemented then. First we lookup a view annotation with max latitude of the bound geometry, view size is not taken into account, e.g.
```
val topAnnotation =
      filteredViewAnnotations.maxByOrNull { (it.geometry!! as Point).latitude() }
```

Then we use this topAnnotation to calculate view bounds and then edge inset
```
(edgeInsets?.top ?: 0).toDouble()
        .plus(abs(getViewAnnotationOptionsFrame(topAnnotation)?.top ?: 0))
```

Am I missing something here?

Reply:
Yeah i see your point here. i will handle this case in https://mapbox.atlassian.net/browse/MAPSAND-577.

Reply:
👍